### PR TITLE
Fix promo code keyboard timer and functionality

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Promo code screen does not reset the screen timeouts](https://trello.com/c/ewLHo1sY/546-promo-code-screen-does-not-reset-the-screen-timeouts)

File: ui/html/choose-fiat.html [hunk 1]
```diff
@@ -22,7 +22,7 @@ <h2 class="js-i18n primary-side-title xs-margin-bottom">How much?</h2>
         <button id="choose-fiat-promo-button" class="bordered-progress-button button d2">
           Have a promo code?
         </button>
-        <p id="choose-fiat-coupon-added" class="d2" style="display:none">✔ Coupon added (5% discount)</p>
+        <p id="choose-fiat-code-added" class="d2" style="display:none">✔ Promo code added (5% discount)</p>
```
Comment (csrapr) [inline ui/html/choose-fiat.html:25]: are these 5% discount messages a placeholder? they seem to be actual hard coded discount rates
there are more examples of this in insert-bills.html and start.html

File: ui/src/app.js [hunk 2]
```diff
@@ -1767,10 +1773,10 @@ function shouldEnableTouch () {
   return chromiumPlus73 || chromePlus73
 }
 
-function setAvailableCoupons (areThereAvailableCoupons) {
-  if (areThereAvailableCoupons) {
-    $('#insert-first-bill-coupon-added').hide()
-    $('#choose-fiat-coupon-added').hide()
+function setAvailablePromoCodes (areThereAvailablePromoCodes) {
```
Comment (csrapr) [inline ui/src/app.js:1776]: why not pull the if inside the else { **if {}** } outside?

like if (a) {} else if (b) else {}

File: ui/src/app.js [hunk 3]
```diff
@@ -1783,14 +1789,14 @@ function setCurrentDiscount (currentDiscount) {
   if (currentDiscount > 0) {
     $('#insert-first-bill-promo-button').hide()
     $('#choose-fiat-promo-button').hide()
-    $('#insert-first-bill-coupon-added').html(`✔ Coupon added (${currentDiscount}% discount)`)
-    $('#choose-fiat-coupon-added').html(`✔ Coupon added (${currentDiscount}% discount)`)
-    $('#insert-first-bill-coupon-added').show()
-    $('#choose-fiat-coupon-added').show()
+    $('#insert-first-bill-code-added').html(`✔ Promo code added (${currentDiscount}% discount)`)
+    $('#choose-fiat-code-added').html(`✔ Promo code added (${currentDiscount}% discount)`)
```
Comment (csrapr) [inline ui/src/app.js:1793]: Rafael said this will break multi language support, "will not generate translation"

File: ui/js/keyboard.js [hunk 4]
```diff
@@ -96,14 +111,33 @@ Keyboard.prototype._keyPress = function _keyPress (target) {
...
+  }
+  // target.addClass('active')
+  // var inputBox = this.inputBox
+  // window.clearTimeout(this.backspaceTimeout)
+  // var content = inputBox.data('content').slice(0, -1)
+  // this.backspaceTimeout = window.setTimeout(function () {
+  //   inputBox.data('content', content)
+  //   inputBox.val(content)
+  //   inputBox.caretToEnd()
+  // }, 100)
```
Comment (csrapr) [inline ui/js/keyboard.js:140]: remove comments

**Review by csrapr — CHANGES_REQUESTED** [review]

File: ui/src/app.js (hunk 3 above)
Comment (chaotixkilla) [reply ui/src/app.js:1793]: It is correct, going to fix that on the PR related to translation

File: ui/html/choose-fiat.html (hunk 1 above)
Comment (chaotixkilla) [reply ui/html/choose-fiat.html:25]: Yup, they are placeholders. Useful for, when developing UI on the machine, having content on the static page

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**